fluent-plugin-prometheus-format/lib/fluent/plugin/filter_prometheus_format.rb

```diff
@@ -0,0 +1,157 @@
+        value_type: :string
+      )
+      config_param(
+        :exclusions,
+        :hash,
+        default: {},
+        symbolize_keys: false,
+        value_type: :string
+      )
+      config_param :strict_inclusions, :bool, default: false
+      config_param :strict_exclusions, :bool, default: false
+      config_param :sort_labels, :bool, default: true
+
+      def configure(conf)
+        super
+        @inclusions.each do |key, value|
+          @inclusions[key] = Regexp.new(value)
+        end
+        @exclusions.each do |key, value|
+          @exclusions[key] = Regexp.new(value)
+        end
+        @metric_accessor = record_accessor_create("$.#{KEY_METRIC}")
+        @timestamp_accessor = record_accessor_create("$.#{KEY_TIMESTAMP}")
+        @value_accessor = record_accessor_create("$.#{KEY_VALUE}")
+      end
+
+      def filter(_tag, _time, record)
+        dotified = dotify_record(record)
+        relabeled = relabel_record(dotified)
+        wrap(to_prometheus_line(relabeled)) if valid?(relabeled)
+      end
+
+      private
+
+      KEY_METRIC = '@metric'.freeze
+      KEY_TIMESTAMP = '@timestamp'.freeze
+      KEY_VALUE = '@value'.freeze
+      KEY_MESSAGE = 'message'.freeze
+      SPLITOR = '.'.freeze
+
+      ORIGIN_KEY = '_origin'.freeze
+      ORIGIN_VALUE = 'kubernetes'.freeze
+
+      def to_prometheus_line(record)
+        metric = @metric_accessor.call(record)
+        timestamp = @timestamp_accessor.call(record)
+        value = @value_accessor.call(record)
+        "#{metric}\{#{to_tags(record)},#{ORIGIN_KEY}=\"#{ORIGIN_VALUE}\"\} #{value} #{timestamp}"
+      end
+
+      def valid?(record)
+        validate_inclusions(record) && validate_exclusions(record)
+      end
+
+      def validate_inclusions(record)
+        @inclusions.each do |key, regex|
+          return false unless validate_inclusion(key, regex, record)
+        end
+        true
+      end
+
+      def validate_inclusion(key, regex, record)
+        if record.key?(key)
+          !regex.match(record[key]).nil?
+        else
+          !@strict_inclusions
+        end
+      end
+
+      def validate_exclusions(record)
+        @exclusions.each do |key, regex|
+          return false unless validate_exclusion(key, regex, record)
+        end
+        true
+      end
+
+      def validate_exclusion(key, regex, record)
+        if record.key?(key)
+          regex.match(record[key]).nil?
+        else
+          !@strict_exclusions
+        end
+      end
+
+      def relabel_record(record)
+        @relabel.flat_map do |org_key, new_key|
+          if record.key?(org_key)
+            value = record[org_key]
+            record.delete(org_key)
+            record[new_key] = value unless new_key.nil? || new_key.empty?
+          end
+        end
+        record
+      end
+
+      def to_tags(hash)
+        array = @sort_labels ? hash.sort : hash.to_a
+        array.map do |key, value|
+          "#{key}=\"#{value}\""\
```

Comment:
Need to escape `\` `"` and `\n` in value
Please also add unit test for these escaping

https://github.com/Sanyaku/sumologic/blob/49f0f94b92a1bcada303f55ab71e1151779246d0/collector-interchange/src/main/scala/com/sumologic/scala/interchange/collector/PrometheusParser.scala#L147